src/features/myAccounts/hooks/useAllSafesGrouped.ts

```diff
@@ -16,18 +16,28 @@ export type AllSafesGrouped = {
   allMultiChainSafes: MultiChainSafeItem[] | undefined
 }
 
-const getMultiChainAccounts = (safes: SafeItems): MultiChainSafeItem[] => {
+export const _buildMultiChainSafeItem = (address: string, safes: SafeItems): MultiChainSafeItem => {
+  const isPinned = safes.some((safe) => safe.isPinned)
+  const lastVisited = safes.reduce((acc, safe) => Math.max(acc, safe.lastVisited || 0), 0)
+  const name = safes.find((safe) => safe.name !== undefined)?.name
```

Comment:
Don't think it makes a huge difference but we could also optimize this by only iterating once instead of three times.

Reply:
Yeah I think it's ok like this